pkg/metrics/metrics.go

```diff
@@ -0,0 +1,275 @@
+	AssignedIdentityDeletionCountM = stats.Int64(assignedIdentityDeletionCountName,
+		"Total number of assigned identity deletion operations",
+		stats.UnitDimensionless)
+
+	// NodeManagedIdentityOperationsDurationM is a measure that tracks the duration in seconds of odemanagedidentity operations.
+	NodeManagedIdentityOperationsDurationM = stats.Float64(
+		nodeManagedIdentityOperationsDurationName,
+		"Duration in seconds of node managed identity operations",
+		stats.UnitMilliseconds)
+	// "operation_type", "status_code"
+
+	// ManagedIdentityControllerCycleDurationM is a measure that tracks the duration in seconds of single cycle in Managed Identity Controller.
+	ManagedIdentityControllerCycleDurationM = stats.Float64(
+		managedIdentityControllerCycleDurationName,
+		"Duration in seconds of single cycle in managed identity controller",
+		stats.UnitMilliseconds)
+
+	// ManagedIdentityControllerCycleCountM is a measure that tracks the cumulative number of cycles executed in managed identity controller.
+	ManagedIdentityControllerCycleCountM = stats.Int64(
+		managedIdentityControllerCycleCountName,
+		"Total number of cycles executed in managed identity controller",
+		stats.UnitDimensionless)
+
+	// ManagedIdentityControllerCycleCountM is a measure that tracks the cumulative number of new leader election in managed identity controller.
+	ManagedIdentityControllerNewLeaderElectionCountM = stats.Int64(
+		managedIdentityControllerNewLeaderElectionCountName,
+		"Total number of new leader election in managed identity controller",
+		stats.UnitDimensionless)
+
+	// CloudProviderOperationsErrorsCountM is a measure that tracks the cumulative number of errors in cloud provider operations.
+	CloudProviderOperationsErrorsCountM = stats.Int64(
+		cloudProviderOperationsErrorsCountName,
+		"Total number of errors in cloud provider operations",
+		stats.UnitDimensionless)
+	// operation_type
+
+	// KubernetesAPIOperationsErrorsCountM is a measure that tracks the cumulative number of errors in cloud provider operations.
+	KubernetesAPIOperationsErrorsCountM = stats.Int64(
+		kubernetesAPIOperationsErrorsCountName,
+		"Total number of errors in kubernetes api operations",
+		stats.UnitDimensionless)
+	// operation_type
+)
+
+var (
+	operationTypeKey = tag.MustNewKey("operation_type")
+	statusCodeKey    = tag.MustNewKey("status_code")
+)
+
+const componentNamespace = "aadpodidentity"
+
+// SinceInSeconds gets the time since the specified start in seconds.
+func SinceInSeconds(start time.Time) float64 {
+	return time.Since(start).Seconds()
+}
+
+// registerViews register views to be collected by exporter
+func registerViews() error {
+
+	views := []*view.View{
+		&view.View{
+			Description: AssignedIdentityAdditionDurationM.Description(),
+			Measure:     AssignedIdentityAdditionDurationM,
+			Aggregation: view.Distribution(0, 1, 2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 97),
```

Comment:
Self review : distribution buckets are too large, as the duration is in seconds. Reduce the same to get more coverage.

Reply:
done.